### ISSUE TITLE
3D printed arms?

### ISSUE THREAD

**Issue by VanushVaswani** [evildmp/BrachioGraph#77]
I love the Lo-Fi nature of this project. Has there been any effort to use 3d printed parts for added stability?

**Comment by ringo2k** [comment]
Hi, i am currently working on 3D printed parts, i will host them on thingiverse when i am finished.
![image](https://user-images.githubusercontent.com/6508358/79369439-4088b100-7f51-11ea-9618-71fddda01c38.png)

**Comment by VanushVaswani** [comment]
Nice, looking forward to that build.

**Comment by ringo2k** [comment]
I uploaded it on thingiverse.. this site did a real downgrade during the last update wow.. 

https://www.thingiverse.com/thing:4295302

Hope it looks better on your side.

**Comment by VanushVaswani** [comment]
Yep. Thingiverse unfortunately sucks now.

Nice work.. have you tried any prints?

**Comment by ringo2k** [comment]
Yes, i make a video and put in on youtube. 
![2020-04-18_15-19 geaendert](https://user-images.githubusercontent.com/6508358/79638878-4a482980-8188-11ea-839e-f4592d266b10.png)


Thingiverse is beeing unusable atm.. sad.

**Comment by VanushVaswani** [comment]
Left a comment on Thingiverse.

**Comment by ringo2k** [comment]
Seen and recommented,

**Comment by oakey1979** [comment]
Sorry to hijack this thread, but I can't get mine working. The various commands to move the servos result in really odd things happening where the arms thrash wildly. Any ideas where I can start troubleshooting?

**Comment by VanushVaswani** [comment]
Did you mount it exactly how the doc said? Show a pic

**Comment by oakey1979** [comment]
![IMG_20200419_110432](https://user-images.githubusercontent.com/63506597/79685086-dceed400-822d-11ea-8f21-d7b41aef81e4.jpg)

**Comment by oakey1979** [comment]
I've gone over the guide and rebuilt and rewired it to double check. You can see from the green lines how it moves when I try to move the arms 5 degrees or draw the box as part of the setup.

**Comment by VanushVaswani** [comment]
Your shoulder servo looks 180 degrees from mine.

**Comment by oakey1979** [comment]
Good spot - I'll rebuild that part, do you think that's the cause of the weird behavior?

**Comment by VanushVaswani** [comment]
Yep

**Comment by VolkerH** [comment]
Looks very nice @ringo2k, will give this a try as there is way too much sag with my ice cream popsicle sticks.

However, the last few times I tried to download the files I see this
![image](https://user-images.githubusercontent.com/2210044/79730109-b1eea800-8333-11ea-87e5-d94b903c9cc5.png)

Not your issue, I will try again.

Thingiverse really has regressed, it would actually be nice to have some Github extensions for 3D prints (simple with OpenSCAD).

**Comment by mcsage** [comment]
@VolkerH Download all files doesn't work but single file download does. https://www.thingiverse.com/thing:4295302/files

**Comment by ringo2k** [comment]
Yes, thingiverse having real troubles the last days.. I uploaded everything three times because there were files missing.. I am still worried that the upload failed, so let me know. I could also but the STL files on my fork, to host it on github?

**Comment by oakey1979** [comment]
I've mounted the shoulder servo the other way around, but I'm still getting crazy behaviour.
See these videos, one is the behaviour just from connecting power, the other from inputting the arm lengths. This also occurs with each subsequent command I use to adjust the angles or draw the box. I've never got any further than this in this project as I have to eventually cut the power when the elbow tries to do a full 360 degree turn and pushes the lifter servo into the shoulder.
https://youtu.be/1rGhffW9mgI
https://youtu.be/ViuSpmldO2g
Any help that can be offered is very welcome!

**Comment by ringo2k** [comment]
@oakey1979 the videos are private, I cannot see them. 

Is your power supply ok? Sometimes servos behave strange because when they drive they need a lot of current, which can result in a voltage drop and then things get weird. How much amps does your power supply handle?

**Comment by oakey1979** [comment]
I've updated the privacy on the videos. In the video you'll see that this is powered from my laptop. It's a Surface 3, I don't know the power output. I have tried plugging in a second USB cable from a mains charging hub to the pi at the same time and had similar problems.

**Comment by ringo2k** [comment]
Oh ok, this really looks strange. When you remove the sticks from the servos, are they still freaking out?

Normally a USB socket can provide 500mA, USB3 900mA. 

I run my Brachiograph with a 3A power supply, a 2A power supply resulted in a restart of the pi. 

Would be interesting to hear what other people are using for powering their build. I mean my Raspi Model 2 is using 300mA in idle, there is no much margin when powered over USB.

**Comment by VanushVaswani** [comment]
Try using a 2.5A supply. Some USB chargers may only have 0.5A

**Comment by ringo2k** [comment]
Hey, I just uploaded my video on youtube. 

https://www.youtube.com/watch?v=Cv_uGYh3t70

**Comment by oakey1979** [comment]
Hi, I've got hold of a powered USB hub and Anker powercore USB cables to ensure there's enough power but my servos are still freaking out like in the videos. I'm about to chuck the whole thing in the bin. Before I do, are there any other suggestions?
